Library/Homebrew/keg.rb

```diff
@@ -309,16 +309,17 @@ def completion_installed?(shell)
     dir && dir.directory? && !dir.children.empty?
   end
 
-  def zsh_functions_installed?
-    # Check for non completion functions (i.e. files not started with an underscore),
-    # since those can be checked separately
-    dir = path.join("share", "zsh", "site-functions")
-    dir && dir.directory? && dir.children.any? { |f| !f.basename.to_s.start_with?("_") }
-  end
-
-  def fish_functions_installed?
-    dir = path.join("share", "fish", "vendor_functions.d")
-    dir && dir.directory? && !dir.children.empty?
+  def functions_installed?(shell)
+    case shell
+    when :fish
+      dir = path.join("share", "fish", "vendor_functions.d")
+      dir && dir.directory? && !dir.children.empty?
+    when :zsh
+      # Check for non completion functions (i.e. files not started with an underscore),
+      # since those can be checked separately
+      dir = path.join("share", "zsh", "site-functions")
+      dir && dir.directory? && dir.children.any? { |f| !f.basename.to_s.start_with?("_") }
```

Comment:
Yeah, I just moved those lines wholesale.  I'll clean them up too, as part of this PR.